docs/michelson_encoder.md

```diff
@@ -533,3 +542,105 @@ const storageSchema = new Schema({
 const mixedSchema = storageSchema.ExtractSchema();
 console.log(JSON.stringify(mixedSchema, null, 2));
 ```
+
+## Breaking Change to Field Numbering {#breaking-change-to-field-numbering}
+When having nested `pair`s or unions (`or`), Taquito assigns numbers to fields when an annotation is not present.
+In previous versions of Taquito, the nested object's fields were numbered were a continuation of the parent object's fields.
+For example, the following schema:
+
+```js live noInline
+const param = {
+    prim: 'or',
+    args: [
+        {
+            prim: 'pair',
+            args: [{ prim: 'address' }, { prim: 'nat' }],
+            annots: ['%transfer']
+        },
+        {
+            prim: 'or',
+            args: [
+                {
+                    prim: 'pair',
+                    args: [{ prim: 'address' }, { prim: 'nat' }],
+                    annots: ['%approve']
+                },
+                {
+                    prim: 'pair',
+                    args: [{ prim: 'address' }, { prim: 'nat' }],
+                    annots: ['%mint']
+                }
+            ]
+        }
+    ],
+    annots: [':_entries']
+};
+const parameterSchema = new ParameterSchema(param);
+Token.fieldNumberingStrategy = "Legacy"; //To bring back the old behavior
+const value = parameterSchema.generateSchema();
+console.log(JSON.stringify(value, null, 2));
+Token.fieldNumberingStrategy = 'Latest'; //To restore the default (new) behavior
+```
+
+Please run the code above and check the output.
+
+Please note how nested field numbers are not predictable. The field numbers are assigned in the order their parent were encountered during the traversal of the tree.
```

Comment:
I feel like these sentences can be improved a little bit, I think I understand them because I have context. Maybe something like

```
Please note that nested field numbers are unpredictable. Field numbers are assigned in the order of where their parent is encountered during the traversal of the tree.
```

What do you think?